### DISCUSSION TITLE
使用VS2022创建新的WinUI项目之后会报错，有时会提示依赖项出现问题，更新了包之后会消失（但有时不会）

### DISCUSSION THREAD

**Discussion by 1100-docx** [microsoft/WindowsAppSDK#4720] [Q&A]
在创建一个新的WinUI项目后会报错：
This version of the Windows App SDK requires Microsoft.Windows.SDK.NET.Ref 10.0.19041.38 or later.
    Please update to .NET SDK 6.0.134, 6.0.426, 8.0.109, 8.0.305 or 8.0.402 (or later).
    Or add a temporary Microsoft.Windows.SDK.NET.Ref reference which can be added with:
        <PropertyGroup>
             <WindowsSdkPackageVersion>10.0.19041.38</WindowsSdkPackageVersion>
        </PropertyGroup>

重新生成解决方案后会有警告提示：
[警告内容.txt](https://github.com/user-attachments/files/17006625/default.txt)

错误提示建议升级 .NET SDK ，但是我的目标框架就是.NET 8.0 或.NET 9.0 了，尝试安装Microsoft.Windows.SDK.NET.Ref包会有错误提示：
包“Microsoft.Windows.SDK.NET.Ref 10.0.26100.45”具有一个包类型“DotnetPlatform”，项目“App6”不支持该类型。
以及错误提示：
包 Microsoft.Windows.SDK.NET.Ref 10.0.26100.45 的包类型 DotnetPlatform 与此项目不兼容。 

并且，在以上方法尝试都失败后，我试着在错误提示的文件（Microsoft.WindowsAppSDK.targets）的代码中添加：
        <PropertyGroup>
             <WindowsSdkPackageVersion>10.0.19041.38</WindowsSdkPackageVersion>
        </PropertyGroup>
后问题依然没有解决

至于依赖项，几乎每一次创建WinUI新项目后都有问题，有时只是包有问题，有时包和程序集都有问题，有时更新一下包就好了，有时更新之后问题依旧

这些问题在8月31日之前是没有的，但是似乎在9月13日我更新了VS2022之后就出现了这些问题，因为这个问题，我已经重新安装了VS2022许多次，并重装了一次系统，都没有解决，这个问题令我很困扰，因为我原本是打算跟我同学一起写一个小程序玩的。
另外，我希望可以在9月16日晚8点前收到答复，万分感谢！

**Comment by 29567474** [comment]
请问解决了吗

**Comment by Riperjame** [comment]
在别的地方看到的,可能会解决：打开项目为项目名称.csproj，在<PropertyGroup></PropertyGroup>添加
<ImplicitUsings>enable</ImplicitUsings>
<Nullable>enable</Nullable>
<SupportedOSPlatformVersion>10.0.22621.0</SupportedOSPlatformVersion>
<WindowsSdkPackageVersion>10.0.22621.45</WindowsSdkPackageVersion>
<img width="795" alt="Snipaste_2024-09-18_17-45-34" src="https://github.com/user-attachments/assets/1cdbcc26-79cc-46d3-a376-50d952da7abf">
还原然后再重新生成解决方案
<img width="1280" alt="Snipaste_2024-09-18_17-49-08" src="https://github.com/user-attachments/assets/f6d52ad4-d059-4809-ab1b-aae2d0bd10b5">

**Reply by ghost** [reply]
正确，可以解决该问题

**Comment by BDRtest1234** [comment]
这个bug我记得以前心血来潮想学习一下，都是正常的，最近重新搞了台电脑，装了VS2022就这毛病

**Comment by huoyaoyuan** [comment]
这是由于.NET SDK和WindowsAppSDK发行不同步导致的已知问题，因为本月.NET SDK没有发布更新而再度延长了一个月。下个月的.NET SDK更新发布后就不再会有这个问题。

**Reply by Linkon-lcw** [reply]
现在有什么办法解决吗

**Reply by huoyaoyuan** [reply]
在你的csproj里添加类似于：

```csproj
  <PropertyGroup>
    <!-- Workaround for CsWinRT 2.1.1. Can be removed after SDK 8.0.402 -->
    <WindowsSdkPackageVersion>10.0.26100.38</WindowsSdkPackageVersion>
  </PropertyGroup>
```

根据你的TargetFramework中的`net8.0-10.0.xxxxx`，把26100替换为你的Windows SDK版本号。

**Reply by Linkon-lcw** [reply]
好的谢谢

**Comment by 1100-docx** [comment]
谢谢您的建议，我将会试一试！很抱歉在最近的十几天里未能回复您，因为我那些天在上学
________________________________
发件人: 29567474 ***@***.***>
发送时间: 2024年9月18日 16:04
收件人: microsoft/WindowsAppSDK ***@***.***>
抄送: 1100-docx ***@***.***>; Author ***@***.***>
主题: Re: [microsoft/WindowsAppSDK] 使用VS2022创建新的WinUI项目之后会报错，有时会提示依赖项出现问题，更新了包之后会消失（但有时不会） (Discussion #4720)


请问解决了吗

―

**Comment by UseCser** [comment]
按理来讲，应该还有一段信息吧 
  Or add a temporary Microsoft.Windows.SDK.NET.Ref reference which can be added with: 

        <PropertyGroup> 

            <WindowsSdkPackageVersion>10.0.19041.38</WindowsSdkPackageVersion> 

        </PropertyGroup> 

在
![image](https://github.com/user-attachments/assets/9d1e2d10-95c2-49db-a6e5-cfe1a8c238d0)
中，打开App1（双击）
在<PropertyGroup></PropertyGroup>中添加<WindowsSdkPackageVersion>10.0.19041.38</WindowsSdkPackageVersion>

**Comment by Super1Windcloud** [comment]
添加了 也是一样报错,无法 识别 

![图片](https://github.com/user-attachments/assets/043d8f44-5690-4fa4-b934-55e50b44fb0a)

**Reply by huoyaoyuan** [reply]
最新版本的VS和.NET SDK早已不需要手动设置
